libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailManagementDialog/useScheduledEmailManagement.ts

```diff
@@ -0,0 +1,64 @@
+// (C) 2022 GoodData Corporation
+
+import { useEffect, useState } from "react";
+import { IScheduledMail } from "@gooddata/sdk-backend-spi";
+import { GoodDataSdkError, useBackendStrict, useWorkspaceStrict } from "@gooddata/sdk-ui";
+import {
+    selectCanManageScheduledMail,
+    selectCurrentUser,
+    selectDashboardRef,
+    useDashboardSelector,
+} from "../../../model";
+
+interface IUseScheduledEmailManagementProps {
+    /**
+     * Callback to be called, when emails fail to load.
+     */
+    onError?: (error: GoodDataSdkError) => void;
+}
+
+export const useScheduledEmailManagement = (props: IUseScheduledEmailManagementProps) => {
+    const dashboardRef = useDashboardSelector(selectDashboardRef);
+    const currentUser = useDashboardSelector(selectCurrentUser);
+    const canManageScheduledMail = useDashboardSelector(selectCanManageScheduledMail);
+    const effectiveBackend = useBackendStrict();
+    const effectiveWorkspace = useWorkspaceStrict();
+
+    const { onError } = props;
+    const [isLoading, setIsLoading] = useState(true);
+    const [scheduledEmails, setScheduledEmails] = useState<IScheduledMail[]>([]);
+
+    useEffect(() => {
```

Comment:
This can be probably simplified with usage of the `useCancelablePromise()` as other hooks fetching backend data

Reply:
Thanks for the suggestion, reworked